core/good_key.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2014 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package core
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"fmt"
+	"reflect"
+	"math/big"
+
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/jose"
+)
+
+// To generate, run: primes 2 752 | tr '\n' ,
+var smallPrimes = []int64{
+	2,3,5,7,11,13,17,19,23,29,31,37,41,43,47,53,59,61,67,71,73,79,83,89,97,101,103,107,109,113,127,131,137,139,149,151,157,163,167,173,179,181,191,193,197,199,211,223,227,229,233,239,241,251,257,263,269,271,277,281,283,293,307,311,313,317,331,337,347,349,353,359,367,373,379,383,389,397,401,409,419,421,431,433,439,443,449,457,461,463,467,479,487,491,499,503,509,521,523,541,547,557,563,569,571,577,587,593,599,601,607,613,617,619,631,641,643,647,653,659,661,673,677,683,691,701,709,719,727,733,739,743,751,
+}
+
+// GoodKey returns true iff the key is acceptable for both TLS use and account
+// key use (our requirements are the same for either one), according to basic
+// strength and algorithm checking.
+func GoodKey(key crypto.PublicKey) bool {
+	log := blog.GetAuditLogger()
+	rsaKey, ok := key.(*rsa.PublicKey)
+	if !ok {
+		if jwk, ok := key.(jose.JsonWebKey); ok && jwk.Rsa != nil {
+			rsaKey = jwk.Rsa
```

Comment:
I'd prefer to go ahead and call a new method like GoodKeyRsa() here and move the below out, and make a stub for GoodKeyEcc() that returns err.